src/main/kotlin/com/zepben/evolve/services/network/tracing/feeder/AssignToLvFeeders.kt

```diff
@@ -25,12 +27,11 @@ class AssignToLvFeeders {
     private val currentTraversal = AssociatedTerminalTrace.newCurrentTrace()
     private lateinit var activeLvFeeder: LvFeeder
 
-    init {
-        normalTraversal.addStepAction(::processNormal)
-        currentTraversal.addStepAction(::processCurrent)
-    }
-
     fun run(network: NetworkService) {
+        val terminalToAuxEquipment = network.sequenceOf<AuxiliaryEquipment>()
+            .filter { it.terminal != null }
```

Comment:
you should just be able to do filterNotNull, which should also get rid of the !! in the groupBy

Reply:
nvm i'm dumb